lib/templates/lab2/lab2.dart

```diff
@@ -1,19 +1,94 @@
-class Book {
-  // your code here
+
+
+
+// import 'dart:io';
+
+class Book with JsonSerializable{
+  final String title;
+  final String author;
+  final int numberOfPages;
+
+  Book(this.title, this.author, this.numberOfPages);
+
+  void bookDetails(){
+    if(title!=null && author!=null && numberOfPages!=null){
+      print("Book\n title: \"$title\"\n author: $author\n pages: $numberOfPages \n");
+    } else {
+      print("Book is not exists\n");
+    }
+  }
 }
 
-class FictionBook {
-  // your code here
+class FictionBook extends Book{
+  String genre;
+
+  FictionBook(String title, String author, int numberOfPages, this.genre) : super(title,author,numberOfPages);
+  
+  @override
+  void bookDetails() {
+    if(title!=null && author!=null && numberOfPages!=null){
+      print("Fiction book\n title: \"$title\"\n author: $author\n genre: $genre\n pages: $numberOfPages\n");
+    } else {
+      print("Book is not exists\n");
+    }
+  }
 }
 
 mixin JsonSerializable {
-  // your code here
+  String? title;
+  String? author;
+  int? numberOfPages;
+  Map<String, dynamic> toJson(){
+    var map = {"title":title, "author":author, "numberOfPages":numberOfPages};
+    return map;
+  }
 }
 
-double calculateArea(){
-  // your code here
+double calculateArea({required double length, required double width, String shape = "Rectangle"}){
```

Comment:
Your default shape in function does not match with case "rectangle" and go through default switch case, beattentive